std/array.d

```diff
@@ -102,22 +102,19 @@ if (isIterable!Range && !isNarrowString!Range && !isInfinite!Range)
     alias E = ForeachType!Range;
     static if (hasLength!Range)
     {
+        E[] result;
         auto length = r.length;
-        if (length == 0)
-            return null;
-
-        import std.conv : emplaceRef;
-
-        auto result = (() @trusted => uninitializedArray!(Unqual!E[])(length))();
 
-        // Every element of the uninitialized array must be initialized
-        size_t i;
-        foreach (e; r)
+        if(length > 0)
         {
-            emplaceRef!E(result[i], e);
-            ++i;
+            result.reserve(length);
+
+            foreach (e; r)
```

Comment:
Ehm, append still going to check capacity on each element. IIRC it's not that fast since it does a round-trip via druntime's extern(C) API. Much as unfortunate as it is I don't believe the _strange_ use case of immutable classes outweights the performance loss.

I say that immutable class is strange for the reason that: 
- due to no being able to express syntactically tail-const object references is of limmited use (very limited at that)
- OOP is typically about incapsulating _mutable_ state inside of objects.
- it contains mutable monitor field anyway